### ISSUE TITLE
Remove gap between pinned and unpinned tabs

### ISSUE THREAD

**Issue by rodrigopedra** [black7375/Firefox-UI-Fix#4]
First of all thanks for this repo. I totally agree with the pro/cons about proton on README.

I added this to my local copy:

~~~css
  #tabbrowser-tabs[haspinnedtabs]:not([positionpinnedtabs]) > #tabbrowser-arrowscrollbox > .tabbrowser-tab[first-visible-unpinned-tab] {
    margin-inline-start: 0 !important;
  }
~~~

It removes the gap between pinned and unpinned tabs. Add it to the repo if you think it is a good addition.

Also, as I use a light theme I needed to tweak line 115 and change the color to `var(--tabs-border-color)` to get tab separators showing. I am not using any additional theme on Firefox, but with the previous value the tab separators won't show.

~~~css
    background-color: var(--tabs-border-color) !important;
~~~

Thanks again and have a nice day.

**Comment by black7375** [comment]
I was hesitant because it seemed like a deliberate addition to the gap from Proton team.
I'm sure compact is, but the rest are still thinking.

A few hours ago, I made an update on the divider line. Would you like to check it?

**Comment by rodrigopedra** [comment]
I started using just before opening issue. Even though I ran a diff and besides my modifications outlined above I am running the last version.

Here are some screenshots:

### Original from repo, no modifications

![image](https://user-images.githubusercontent.com/5470108/116992315-33e61080-acac-11eb-80c1-daa419c3282d.png)

### With only border color modification

![image](https://user-images.githubusercontent.com/5470108/116992445-66900900-acac-11eb-9935-f0ad22c9344d.png)

### With both border color and tab gap modifications

![image](https://user-images.githubusercontent.com/5470108/116992238-1ca72300-acac-11eb-9f19-4a2256d3305c.png)


One thing to correct from my first post: line to be modified to use `var(--tabs-border-color)` should be line 113, not line 115 as I first stated. Reason I had an offset is that I added a comment on the top of my `userChrome.css` to keep a reference to this repo's URL.

If I can provide any other info, let me know.

**Comment by rodrigopedra** [comment]
Here is the diff between the original file and my custom file:

![image](https://user-images.githubusercontent.com/5470108/116993046-4ca2f600-acad-11eb-9411-f2db4f14cb56.png)

**Comment by black7375** [comment]
Nice report!!😎
I'm lying down to sleep now, so I'll fix it in the morning.

**Comment by rodrigopedra** [comment]
You’re welcome! No rush on this, take your time and sleep tight. 

Thanks again for the nice solution

**Comment by black7375** [comment]
### Remove Gaps

First of all, I made it active in compact mode.
https://github.com/black7375/Firefox-UI-Fix/commit/fa74a2ba332fe081d007782797848d28330d1b88

### Seperator Test

Extensive testing was done for the unselected tab's seperator.
**Current Version**
![firefox_proton_my_tab_seperator1](https://user-images.githubusercontent.com/25581533/117103664-4aeb3800-ad6a-11eb-809a-fa84f4b35c3c.png)
![firefox_proton_my_tab_seperator2](https://user-images.githubusercontent.com/25581533/117103666-4c1c6500-ad6a-11eb-96bf-04cd43960968.png)
![firefox_proton_my_tab_seperator3](https://user-images.githubusercontent.com/25581533/117103668-4cb4fb80-ad6a-11eb-9efe-472323d6aaa5.png)
![firefox_proton_my_tab_seperator4](https://user-images.githubusercontent.com/25581533/117103669-4cb4fb80-ad6a-11eb-89a0-5c7585fd96fc.png)
![firefox_proton_my_tab_seperator5](https://user-images.githubusercontent.com/25581533/117103671-4d4d9200-ad6a-11eb-80c3-bbdab7bf53bf.png)
![firefox_proton_my_tab_seperator6](https://user-images.githubusercontent.com/25581533/117103672-4d4d9200-ad6a-11eb-9761-49b8144b9e41.png)

- Proton Light Theme
- Proton Dark Theme
- Firefox Alpenglow [Light]
- Firefox Alpenglow [Dark]
- System Default [Light]
- System Default [Dark]

**Suggestion Version**
![firefox_proton_issue_tab_seperator1](https://user-images.githubusercontent.com/25581533/117103728-65251600-ad6a-11eb-8ef5-697f73290b9b.png)
![firefox_proton_issue_tab_seperator2](https://user-images.githubusercontent.com/25581533/117103731-65bdac80-ad6a-11eb-82db-a52668139ea0.png)
![firefox_proton_issue_tab_seperator3](https://user-images.githubusercontent.com/25581533/117103732-65bdac80-ad6a-11eb-8053-51e1d8a2adab.png)
![firefox_proton_issue_tab_seperator4](https://user-images.githubusercontent.com/25581533/117103734-66eed980-ad6a-11eb-9251-d3f9eb154eb4.png)
![firefox_proton_issue_tab_seperator5](https://user-images.githubusercontent.com/25581533/117103739-67877000-ad6a-11eb-9108-b085f41e7e8a.png)
![firefox_proton_issue_tab_seperator6](https://user-images.githubusercontent.com/25581533/117103741-67877000-ad6a-11eb-81b6-9bd6c462afca.png)

- Proton Light Theme
- Proton Dark Theme
- Firefox Alpenglow [Light]
- Firefox Alpenglow [Dark]
- System Default [Light]
- System Default [Dark]

**More GTK Themes**
- Adwaita
- Adwaita-dark
- Breath[Screenshot's light system Theme]
- Breath-Dark[Screenshot's dark system Theme]
- Default
- Emacs
- HighContrast


### Result

It was `--toolbarseparator-color` that looked consistently well. (Current Version).
But I'v been certainly knew that some themes (including GTK) may not be visible.

Therefore, we have created a new wiki document dealing with compatibility issues!!
https://github.com/black7375/Firefox-UI-Fix/wiki/Compatibility-Issues-Solution

**Comment by rodrigopedra** [comment]
Thanks for the wiki entry.

I had to keep `--tabs-border-color` in order to have the borders visible.

One thing to correct, although I am using KDE Plasma 5.21 with stock Breeze Twilight theme (from your screenshots it seems you are running KDE Plasma as well), I am using a custom color scheme called Ambiance-ISH (link below)

https://store.kde.org/p/1001434

Of course as this is a custom color scheme, I don't expect it to be handled by the original `userChrome.css` file provided here.

Totally forgot to mention it before, sorry. I use this color theme for so long I usually forget about it.

Also I kept the gap removal from my first comment, but also understand your reasoning around it. But maybe you could add it to the week to those who might want to remove it.

Thanks for all the help and the hard work to provide this custom CSS.

**Comment by black7375** [comment]
I will add an issue template that asks about the theme in use.

\+ Good News
I think we'll fix the icon problem around next week. https://github.com/black7375/Firefox-UI-Fix/issues/2#issuecomment-832572470

**Comment by black7375** [comment]
Actually, I hesitated because of the compatibility issues of tab separators and only provided it in the compact.

I now believe that a lot has been resolved, so we will remove the gap.
Thanks.
